main.cpp

```diff
@@ -74,8 +74,8 @@ typedef steady_clock::time_point time_point;
 
 const int print_stats_interval = 60;
 const int nodes_in_response = 16;
-int node_buffer_size = 10000000;
-int ping_queue_size = 5000000;
+size_t node_buffer_size = 10000000;
```

Comment:
this makes me a bit nervous. is there a good reason to use unsigned types here? my experience is that using unsigned integers is more error prone than signed.